core/src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java

```diff
@@ -415,8 +417,8 @@ private boolean isChanged(
         || latestResult.get().getVersion() != result.get().getVersion();
   }
 
-  private void throwExceptionDueToPotentialAntiDependency() throws CommitConflictException {
-    throw new CommitConflictException(
+  private void throwExceptionDueToPotentialAntiDependency() throws PreparationConflictException {
```

Comment:
Made this method throw `PreparationConflictException` instead of `CommitConflictException`.